spec/spec_helper/index.rb

```diff
@@ -78,10 +78,91 @@ def sort_dependencies(dependencies, activated, conflicts)
         name = name_for(dependency)
         [
           activated.vertex_named(name).payload ? 0 : 1,
+          amount_constrained(dependency),
           conflicts[name] ? 0 : 1,
           activated.vertex_named(name).payload ? 0 : search_for(dependency).count,
         ]
       end
     end
+
+    def amount_constrained(dependency)
+      @amount_constrained ||= {}
+      @amount_constrained[dependency.name] ||= begin
+        all = specs[dependency.name].size
+        if all <= 1
+          all #- 1_000_000
+        else
+          search = search_for(dependency).size
+          search - all
+        end
+      end
+    end
   end
+
+  class ReverseBundlerIndex < BundlerIndex
+    def sort_dependencies(*)
+      super.reverse
+    end
+  end
+
+  class RandomSortIndex < TestIndex
+    def sort_dependencies(dependencies, _, _)
```

Comment:
This fails on ruby 1.8.7 (the cause of the CI failure)

```
SyntaxError: compile error
(irb):1: duplicate argument name
        from (irb):2
```

Maybe it can be `def sort_dependencies(dependencies, _activated, _conflicts)`